src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilderRow.tsx

```diff
@@ -39,61 +46,95 @@ const QueryBuilderRow: React.FC<IQueryBuilderRowProps> = (props) => {
         onChangeValue,
         position,
         removeRow,
-        styles
+        styles,
+        theme,
+        updateSnippet
     } = props;
+    const propertySelectorId = useId('cb-advanced-search-property-select');
     const classNames = getClassNames(styles, {
-        theme: useTheme()
+        theme: useTheme(),
+        isOnlyFirstRow: isRemoveDisabled
     });
+    const { t } = useTranslation();
 
-    const [comboBoxOptions, setComboBoxOptions] = useState<IComboBoxOption[]>(
-        []
+    /** React select values */
+    const { isLoading, flattenedModelProperties } = useFlattenedModelProperties(
+        {
+            adapter,
+            allowedPropertyValueTypes
+        }
     );
+
+    const [propertyOptions, setPropertyOptions] = useState([]);
+
+    // Other
     const [selectedOperator, setSelectedOperator] = useState<
         IDropdownOption<OperatorData>
     >();
     const [operatorOptions, setOperatorOptions] = useState<
         IDropdownOption<OperatorData>[]
     >();
-    const [selectedProperty, setSelectedProperty] = useState<IDropdownOption>();
+    const [selectedProperty, setSelectedProperty] = useState<PropertyOption>();
     const [selectedCombinator, setSelectedCombinator] = useState<string>();
+    const [selectedValue, setSelectedValue] = useState<string>();
 
-    const { isLoading, flattenedProperties } = useFlattenedProperties({
-        adapter,
-        allowedPropertyValueTypes
-    });
-
+    // Effects
+    // Creating react-select options
     useEffect(() => {
-        const comboBoxOptions: IComboBoxOption[] = [];
-        if (flattenedProperties) {
-            Object.keys(flattenedProperties).forEach((modelName: string) => {
-                comboBoxOptions.push({
-                    key: modelName,
-                    text: modelName,
-                    itemType: SelectableOptionMenuItemType.Header
-                });
-                flattenedProperties[modelName].forEach((property) => {
-                    comboBoxOptions.push({
-                        key: property.key,
-                        text: property.localPath,
-                        data: {
-                            name: property.name,
-                            type: property.propertyType
-                        }
+        const propertyComboboxOptions = [];
+        if (flattenedModelProperties) {
+            Object.keys(flattenedModelProperties).forEach(
```

Comment:
suggestion: could pull the creation logic out to a module scope helper that returns the list of options to keep render a bit more condensed